### ISSUE TITLE
Typescript check not passing after Deno 1.13.2 

### ISSUE THREAD

**Issue by jintaokoong** [eveningkid/denodb#303]
### Environment
**OS**: Linux 5.10.70-1-MANJARO
**Deno version**: deno 1.14.0 
**V8 version**: v8 9.4.146.15
**TS version**: typescript 4.4.2

### Issue
Importing denodb causes TS check to not pass after deno 1.13.2. 

code
---
```typescript
import { DataTypes, Database, Model, PostgresConnector } from 'https://deno.land/x/denodb/mod.ts';
```

outputs
---
```
error: TS2578 [ERROR]: Unused '@ts-expect-error' directive.
      // @ts-expect-error
      ~~~~~~~~~~~~~~~~~~~
    at https://deno.land/std@0.106.0/node/events.ts:100:7

TS2578 [ERROR]: Unused '@ts-expect-error' directive.
        // @ts-expect-error
        ~~~~~~~~~~~~~~~~~~~
    at https://deno.land/std@0.106.0/node/events.ts:105:9

TS2578 [ERROR]: Unused '@ts-expect-error' directive.
      // @ts-expect-error
      ~~~~~~~~~~~~~~~~~~~
    at https://deno.land/std@0.106.0/node/events.ts:116:7

TS2578 [ERROR]: Unused '@ts-expect-error' directive.
      // @ts-expect-error
      ~~~~~~~~~~~~~~~~~~~
    at https://deno.land/std@0.106.0/node/events.ts:200:7

TS2578 [ERROR]: Unused '@ts-expect-error' directive.
    // @ts-expect-error
    ~~~~~~~~~~~~~~~~~~~
    at https://deno.land/std@0.106.0/node/events.ts:225:5

TS2578 [ERROR]: Unused '@ts-expect-error' directive.
        // @ts-expect-error
        ~~~~~~~~~~~~~~~~~~~
    at https://deno.land/std@0.106.0/node/events.ts:391:9

TS2578 [ERROR]: Unused '@ts-expect-error' directive.
      // @ts-expect-error
      ~~~~~~~~~~~~~~~~~~~
    at https://deno.land/std@0.106.0/node/events.ts:424:7

TS2578 [ERROR]: Unused '@ts-expect-error' directive.
          // @ts-expect-error
          ~~~~~~~~~~~~~~~~~~~
    at https://deno.land/std@0.106.0/node/events.ts:446:11

TS2578 [ERROR]: Unused '@ts-expect-error' directive.
          // @ts-expect-error
          ~~~~~~~~~~~~~~~~~~~
    at https://deno.land/std@0.106.0/node/events.ts:451:11

TS2578 [ERROR]: Unused '@ts-expect-error' directive.
    // @ts-expect-error
    ~~~~~~~~~~~~~~~~~~~
    at https://deno.land/std@0.106.0/node/events.ts:646:5

TS2578 [ERROR]: Unused '@ts-expect-error' directive.
    // @ts-expect-error
    ~~~~~~~~~~~~~~~~~~~
    at https://deno.land/std@0.106.0/node/events.ts:667:5

Found 11 errors.
```

---

I suppose this is caused by one of the dependencies that depends on the std library, maybe bumping the offending dependency would resolve this issue?

**Comment by seanmavley** [comment]
I can't seem to use it, because this error keeps coming up. Is the only workaround to downgrade to a version earlier than the above?

**Comment by kinghat** [comment]
now also in 1.16.0 we get:
> error: TS2339 [ERROR]: Property 'getIterator' does not exist on type 'ReadableStream<R>'.
  return res.readable.getIterator();

i think its in the mongo module:
<pre>
│ ├─┬ https://deno.land/x/denodb@v1.0.39/mod.ts (1002B)
│ │ ├─┬ https://deno.land/x/denodb@v1.0.39/lib/connectors/connector.ts (1.25KB)
│ │ │ ├─┬ https://deno.land/x/denodb@v1.0.39/lib/query-builder.ts (5.54KB)
│ │ │ │ ├─┬ https://deno.land/x/denodb@v1.0.39/deps.ts (1.35KB)
│ │ │ │ │ ├─┬ https://deno.land/x/mongo@v0.22.0/mod.ts (113B)
│ │ │ │ │ │ ├─┬ https://deno.land/x/mongo@v0.22.0/deps.ts (783B)
│ │ │ │ │ │ │ ├─┬ https://deno.land/std@0.83.0/node/_crypto/pbkdf2.ts (4.18KB)
│ │ │ │ │ │ │ │ ├── https://deno.land/std@0.83.0/hash/mod.ts *
│ │ │ │ │ │ │ │ ├─┬ https://deno.land/std@0.83.0/node/buffer.ts (15.46KB)
│ │ │ │ │ │ │ │ │ ├── https://deno.land/std@0.83.0/encoding/base64.ts *
│ │ │ │ │ │ │ │ │ ├── https://deno.land/std@0.83.0/encoding/hex.ts *
│ │ │ │ │ │ │ │ │ └─┬ https://deno.land/std@0.83.0/node/_utils.ts (5.74KB)
│ │ │ │ │ │ │ │ │   ├─┬ https://deno.land/std@0.83.0/async/mod.ts (202B)
│ │ │ │ │ │ │ │ │   │ ├── https://deno.land/std@0.83.0/async/deferred.ts *
│ │ │ │ │ │ │ │ │   │ ├── https://deno.land/std@0.83.0/async/delay.ts (279B)
│ │ │ │ │ │ │ │ │   │ ├─┬ https://deno.land/std@0.83.0/async/mux_async_iterator.ts (1.96KB)
│ │ │ │ │ │ │ │ │   │ │ └── https://deno.land/std@0.83.0/async/deferred.ts *
<b>│ │ │ │ │ │ │ │ │   │ └── https://deno.land/std@0.83.0/async/pool.ts (1.58KB)</b>
</pre>

**Comment by kinghat** [comment]
> I can't seem to use it, because this error keeps coming up. Is the only workaround to downgrade to a version earlier than the above?

you could use an import map like this:
```json
{
  "imports": {
    "https://deno.land/std@0.106.0/node/events.ts": "https://deno.land/std@0.114.0/node/events.ts",
    "https://deno.land/std@0.83.0/async/pool.ts": "https://deno.land/std@0.114.0/async/pool.ts"
  }
}
```

**Comment by lejacobroy** [comment]
I think I narowed-down the issue.
It comes from the 'dex' dependency, pointing at the master branch.
`export { default as SQLQueryBuilder } from "https://raw.githubusercontent.com/aghussb/dex/master/mod.ts";`

Could we point to a previous fixed release instead?

**Comment by SammyShehter** [comment]
deno 1.16.1 Same issue

**Comment by haze5959** [comment]
update please
help

**Comment by Zhomart** [comment]
Still happening. 

```
deno 1.16.2 (release, x86_64-apple-darwin)
v8 9.7.106.2
typescript 4.4.2
```


import map didn't help for me unfortunately. What other people are doing?

```
thread 'main' panicked at 'internal error: entered unreachable code: unexpected missing emit: https://deno.land/x/mongo@v0.22.0/src/collection/commands/aggregate.ts', cli/proc_state.rs:544:15
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
```

**Comment by demarche1** [comment]
deno 1.16.2 Same problem.
I managed to use it using version 1.15.3

**Comment by Zhomart** [comment]
This pull #307 solves the compatibility issues with the latest deno (v.1.16.2).

The problem was with the dex library as @lejacobroy pointed out above, postgres lib, mysql, mongodb and sqlite dependencies.

I haven't tested the PR on serious projects.

**Comment by dada513** [comment]
Until #307 gets merged, a workaround is importing the PR directly:
```ts
import { Database } from "https://raw.githubusercontent.com/Zhomart/denodb/update-deps/mod.ts";
```

**Comment by eveningkid** [comment]
This should be fixed with #307!

Please upgrade to `master` or `1.0.40`.

Have a great day :)
